presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcMetadata.java

```diff
@@ -93,13 +93,9 @@ public JdbcTableHandle getTableHandle(ConnectorSession session, SchemaTableName
             return Optional.empty();
         }
 
-        handle = new JdbcTableHandle(
-                handle.getSchemaTableName(),
-                handle.getCatalogName(),
-                handle.getSchemaName(),
-                handle.getTableName(),
-                newDomain,
-                handle.getLimit());
+        handle = JdbcTableHandle.builder(handle)
+                .setConstraint(newDomain)
+                .build();
```

Comment:
This place doesn't really benefit from the fact JdbcTableHandle has a builder, because we don't know whether it's safe to add a constraint -- this depends what fields are already set and what's their evaluation order.

in fact, the problem already exist, as we apply constraint even if the limit is already set, which isn't correct (changes semantics)

from this PR perspective, it's safer to use constructor directly, so that we're forced to update this site when we add new field to the handle class

Reply:
(we could address this issue by providing`withConstraint(TupleDomain)` method which would would throw when limit is already set)

Reply:
Reverted to use the ctor directly.